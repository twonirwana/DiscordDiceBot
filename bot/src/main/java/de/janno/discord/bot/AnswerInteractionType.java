package de.janno.discord.bot;

public enum AnswerInteractionType {
    none,
    hidden,
    reroll
}
