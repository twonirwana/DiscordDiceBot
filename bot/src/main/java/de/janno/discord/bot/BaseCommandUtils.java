package de.janno.discord.bot;

import de.janno.discord.bot.persistance.Mapper;
import de.janno.discord.bot.persistance.MessageDataDTO;
import de.janno.discord.bot.persistance.PersistenceManager;
import lombok.NonNull;

import javax.annotation.Nullable;
import java.util.UUID;

public class BaseCommandUtils {

    public static MessageDataDTO createEmptyMessageData(@NonNull UUID configUUID,
                                                        @Nullable Long guildId,
                                                        long channelId,
                                                        long messageId,
                                                        String commandId,
                                                        PersistenceManager persistenceManager) {
        MessageDataDTO messageDataDTO = new MessageDataDTO(configUUID, guildId, channelId, messageId, commandId, Mapper.NO_PERSISTED_STATE, null);
        //should not be needed but sometimes there is a retry ect and then there is already a state
        persistenceManager.deleteStateForMessage(channelId, messageId);
        persistenceManager.saveMessageData(messageDataDTO);
        return messageDataDTO;
    }
}
