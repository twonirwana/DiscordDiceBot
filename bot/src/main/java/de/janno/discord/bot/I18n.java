package de.janno.discord.bot;

import com.google.common.collect.ImmutableList;
import de.janno.discord.connector.api.slash.CommandLocaleChoice;
import de.janno.discord.connector.api.slash.CommandLocaleDescription;
import de.janno.discord.connector.api.slash.CommandLocaleName;
import lombok.extern.slf4j.Slf4j;
import org.apache.commons.text.StringEscapeUtils;

import java.text.MessageFormat;
import java.util.*;

@Slf4j
public final class I18n {
    private final static String MESSAGES_KEY = "botMessages";

    private I18n() {
    }

    public static String getMessage(String key, Locale locale) {
        return StringEscapeUtils.unescapeJava(ResourceBundle.getBundle(MESSAGES_KEY, locale).getString(key));
    }

    public static String getMessage(String key, Locale locale, Object... arguments) {
        return MessageFormat.format(getMessage(key, locale), arguments);
    }

    public static List<Locale> getAdditionalLanguage() {
        return List.of(Locale.GERMAN, Locale.of("pt","BR"));
    }

    public static List<Locale> allSupportedLanguage() {
        return ImmutableList.<Locale>builder()
                .add(Locale.ENGLISH)
                .addAll(getAdditionalLanguage())
                .build();
    }

    private static List<LocaleString> allNoneEnglishMessagesValues(String key) {
        return getAdditionalLanguage().stream()
                .map(l -> new LocaleString(l, getMessage(key, l)))
                .filter(m -> !Objects.equals(m.value(), getMessage(key, Locale.ENGLISH))) //remove all locals that are equal to the default english one
                .toList();
    }

    public static List<CommandLocaleName> allNoneEnglishMessagesNames(String key) {
        return allNoneEnglishMessagesValues(key).stream()
                .map(lv -> new CommandLocaleName(lv.locale(), lv.value()))
                .toList();
    }

    public static List<CommandLocaleDescription> allNoneEnglishMessagesDescriptions(String key) {
        return allNoneEnglishMessagesValues(key).stream()
                .map(lv -> new CommandLocaleDescription(lv.locale(), lv.value()))
                .toList();
    }

    public static List<CommandLocaleChoice> allNoneEnglishMessagesChoices(String key) {
        return allNoneEnglishMessagesValues(key).stream()
                .map(lv -> new CommandLocaleChoice(lv.locale(), lv.value()))
                .toList();
    }

    public static List<CommandLocaleDescription> allNoneEnglishDescriptionsWithKeys(String key, String... keys) {
        return getAdditionalLanguage().stream()
                .map(l -> new CommandLocaleDescription(l, getMessage(key, l, Arrays.stream(keys)
                        .map(k -> getMessage(k, l))
                        .toArray(Object[]::new))))
                .filter(m -> !Objects.equals(m.getDescription(), getMessage(key, Locale.ENGLISH))) //remove all locals that are equal to the default english one
                .toList();
    }

    private record LocaleString(Locale locale, String value){}

}
