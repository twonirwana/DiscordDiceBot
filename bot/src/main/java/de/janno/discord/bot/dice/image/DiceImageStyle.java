package de.janno.discord.bot.dice.image;

import de.janno.discord.bot.dice.image.provider.*;

import java.util.Arrays;
import java.util.List;
import java.util.Map;
import java.util.stream.Collectors;

public enum DiceImageStyle {
    none(new NoneProvider()),
    polyhedral_3d(new PolyhedralFileImageProvider("polyhedral_3d", List.of("red_and_white"), "red_and_white")),
    polyhedral_alies_v2(new PolyhedralFileImageProvider(
            "polyhedral_alies_v2",
            List.of("blue_and_silver", "black_and_gold", "blue_and_gold", "green_and_gold", "orange_and_silver", "red_and_gold", "purple_and_silver"),
            "blue_and_silver")),
    polyhedral_knots(new PolyhedralFileImageProvider("polyhedral_knots", List.of("purple_dark", "purple_white", "blue"), "blue")),
    polyhedral_RdD(new PolyhedralFileImageProvider("polyhedral_RdD", List.of("default", "special"), "default", Map.of(
            "default", List.of(4, 6, 7, 8, 10, 12, 20, 100),
            "special", List.of(8, 12)))),
    fate(new FateBasic()),
    d6_dots(new D6Dotted()),
    polyhedral_2d(new PolyhedralSvgWithColor()),
    polyhedral_alies_v1(new PolyhedralFileImageProvider("polyhedral_alies_v1", List.of("black_and_gold"), "black_and_gold"));

    private final ImageProvider imageProvider;

    DiceImageStyle(ImageProvider imageProvider) {
        this.imageProvider = imageProvider;
    }

    public static boolean isValidStyle(String name) {
        return Arrays.stream(values()).map(Enum::name).collect(Collectors.toSet()).contains(name);
    }

    public static String combineStyleAndColorName(DiceImageStyle style, String color) {
        return "%s_%s".formatted(style.name(), color);
    }

    public ImageProvider getImageProvider() {
        return imageProvider;
    }

    public String getDefaultColor() {
        return imageProvider.getDefaultColor();
    }

    public List<String> getSupportedColors() {
        return imageProvider.getSupportedColors();
    }
}
