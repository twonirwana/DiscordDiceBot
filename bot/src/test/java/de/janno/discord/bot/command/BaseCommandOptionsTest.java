package de.janno.discord.bot.command;

import au.com.origin.snapshots.Expect;
import au.com.origin.snapshots.junit5.SnapshotExtension;
import de.janno.discord.bot.I18n;
import de.janno.discord.bot.dice.image.DiceImageStyle;
import de.janno.discord.connector.api.AutoCompleteAnswer;
import de.janno.discord.connector.api.AutoCompleteRequest;
import de.janno.discord.connector.api.OptionValue;
import org.junit.jupiter.api.extension.ExtendWith;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

import java.util.Arrays;
import java.util.List;
import java.util.Locale;
import java.util.stream.Stream;

@ExtendWith(SnapshotExtension.class)
class BaseCommandOptionsTest {

    Expect expect;

    static Stream<Arguments> generateColorLocaleData() {
        return Arrays.stream(DiceImageStyle.values())
                .flatMap(d -> I18n.allSupportedLanguage().stream()
                        .map(l -> Arguments.of(d, l)));
    }

    static Stream<Arguments> generateLocaleData() {
        return I18n.allSupportedLanguage().stream()
                .map(Arguments::of);
    }

    @ParameterizedTest
    @MethodSource("generateColorLocaleData")
    void autoCompleteColorOption(DiceImageStyle style, Locale userLocal) {
        List<AutoCompleteAnswer> res = BaseCommandOptions.autoCompleteColorOption(new AutoCompleteRequest("dice_image_color", "", List.of(new OptionValue("dice_image_style", style.name()))), userLocal);
        expect.scenario(style + "_" + userLocal).toMatchSnapshot(res);
    }

    @ParameterizedTest
    @MethodSource("generateLocaleData")
    void autoCompleteColorOption_invalidStyle(Locale userLocal) {
        List<AutoCompleteAnswer> res = BaseCommandOptions.autoCompleteColorOption(new AutoCompleteRequest("dice_image_color", "", List.of(new OptionValue("dice_image_style", "test"))), userLocal);
        expect.scenario(userLocal.toString()).toMatchSnapshot(res);
    }
}