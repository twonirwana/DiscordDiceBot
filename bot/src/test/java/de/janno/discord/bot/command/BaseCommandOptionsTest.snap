de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_dots_de]=[
  [
    {
      "name": "weiß",
      "value": "white"
    },
    {
      "name": "schwarz_und_gold",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_dots_en]=[
  [
    {
      "name": "white",
      "value": "white"
    },
    {
      "name": "black_and_gold",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_dots_fr]=[
  [
    {
      "name": "blanc",
      "value": "white"
    },
    {
      "name": "noir_et_or",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_dots_pt_BR]=[
  [
    {
      "name": "branco",
      "value": "white"
    },
    {
      "name": "preto_e_dourado",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_marvel_de]=[
  [
    {
      "name": "blau",
      "value": "blue"
    },
    {
      "name": "rot",
      "value": "red"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_marvel_en]=[
  [
    {
      "name": "blue",
      "value": "blue"
    },
    {
      "name": "red",
      "value": "red"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_marvel_fr]=[
  [
    {
      "name": "bleu",
      "value": "blue"
    },
    {
      "name": "rouge",
      "value": "red"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[d6_marvel_pt_BR]=[
  [
    {
      "name": "azul",
      "value": "blue"
    },
    {
      "name": "vermelho",
      "value": "red"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[expanse_de]=[
  [
    {
      "name": "gürtel_dunkel",
      "value": "belt_dark"
    },
    {
      "name": "gürtel_hell",
      "value": "belt_light"
    },
    {
      "name": "erde_dunkel",
      "value": "earth_dark"
    },
    {
      "name": "erde_hell",
      "value": "earth_light"
    },
    {
      "name": "mars_dunkel",
      "value": "mars_dark"
    },
    {
      "name": "mars_hell",
      "value": "mars_light"
    },
    {
      "name": "protogen_dunkel",
      "value": "protogen_dark"
    },
    {
      "name": "protogen_hell",
      "value": "protogen_light"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[expanse_en]=[
  [
    {
      "name": "belt_dark",
      "value": "belt_dark"
    },
    {
      "name": "belt_light",
      "value": "belt_light"
    },
    {
      "name": "earth_dark",
      "value": "earth_dark"
    },
    {
      "name": "earth_light",
      "value": "earth_light"
    },
    {
      "name": "mars_dark",
      "value": "mars_dark"
    },
    {
      "name": "mars_light",
      "value": "mars_light"
    },
    {
      "name": "protogen_dark",
      "value": "protogen_dark"
    },
    {
      "name": "protogen_light",
      "value": "protogen_light"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[expanse_fr]=[
  [
    {
      "name": "ceinture_foncée",
      "value": "belt_dark"
    },
    {
      "name": "ceinture_claire",
      "value": "belt_light"
    },
    {
      "name": "terre_sombre",
      "value": "earth_dark"
    },
    {
      "name": "terre_claire",
      "value": "earth_light"
    },
    {
      "name": "mars_sombre",
      "value": "mars_dark"
    },
    {
      "name": "mars_clair",
      "value": "mars_light"
    },
    {
      "name": "protogen_dark",
      "value": "protogen_dark"
    },
    {
      "name": "protogen_clair",
      "value": "protogen_light"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[expanse_pt_BR]=[
  [
    {
      "name": "belt_dark",
      "value": "belt_dark"
    },
    {
      "name": "belt_light",
      "value": "belt_light"
    },
    {
      "name": "earth_dark",
      "value": "earth_dark"
    },
    {
      "name": "earth_light",
      "value": "earth_light"
    },
    {
      "name": "mars_dark",
      "value": "mars_dark"
    },
    {
      "name": "mars_light",
      "value": "mars_light"
    },
    {
      "name": "protogen_dark",
      "value": "protogen_dark"
    },
    {
      "name": "protogen_light",
      "value": "protogen_light"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[fate_de]=[
  [
    {
      "name": "schwarz",
      "value": "black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[fate_en]=[
  [
    {
      "name": "black",
      "value": "black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[fate_fr]=[
  [
    {
      "name": "noir",
      "value": "black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[fate_pt_BR]=[
  [
    {
      "name": "preto",
      "value": "black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[none_de]=[
  [
    {
      "name": "kein",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[none_en]=[
  [
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[none_fr]=[
  [
    {
      "name": "aucun",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[none_pt_BR]=[
  [
    {
      "name": "nenhum",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_2d_de]=[
  [
    {
      "name": "weiß",
      "value": "white"
    },
    {
      "name": "grau",
      "value": "gray"
    },
    {
      "name": "schwarz",
      "value": "black"
    },
    {
      "name": "rot",
      "value": "red"
    },
    {
      "name": "pink",
      "value": "pink"
    },
    {
      "name": "orange",
      "value": "orange"
    },
    {
      "name": "gelb",
      "value": "yellow"
    },
    {
      "name": "grün",
      "value": "green"
    },
    {
      "name": "magenta",
      "value": "magenta"
    },
    {
      "name": "cyan",
      "value": "cyan"
    },
    {
      "name": "blau",
      "value": "blue"
    },
    {
      "name": "indigo",
      "value": "indigo"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_2d_en]=[
  [
    {
      "name": "white",
      "value": "white"
    },
    {
      "name": "gray",
      "value": "gray"
    },
    {
      "name": "black",
      "value": "black"
    },
    {
      "name": "red",
      "value": "red"
    },
    {
      "name": "pink",
      "value": "pink"
    },
    {
      "name": "orange",
      "value": "orange"
    },
    {
      "name": "yellow",
      "value": "yellow"
    },
    {
      "name": "green",
      "value": "green"
    },
    {
      "name": "magenta",
      "value": "magenta"
    },
    {
      "name": "cyan",
      "value": "cyan"
    },
    {
      "name": "blue",
      "value": "blue"
    },
    {
      "name": "indigo",
      "value": "indigo"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_2d_fr]=[
  [
    {
      "name": "blanc",
      "value": "white"
    },
    {
      "name": "gris",
      "value": "gray"
    },
    {
      "name": "noir",
      "value": "black"
    },
    {
      "name": "rouge",
      "value": "red"
    },
    {
      "name": "rose",
      "value": "pink"
    },
    {
      "name": "orange",
      "value": "orange"
    },
    {
      "name": "jaune",
      "value": "yellow"
    },
    {
      "name": "vert",
      "value": "green"
    },
    {
      "name": "magenta",
      "value": "magenta"
    },
    {
      "name": "cyan",
      "value": "cyan"
    },
    {
      "name": "bleu",
      "value": "blue"
    },
    {
      "name": "indigo",
      "value": "indigo"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_2d_pt_BR]=[
  [
    {
      "name": "branco",
      "value": "white"
    },
    {
      "name": "cinza",
      "value": "gray"
    },
    {
      "name": "preto",
      "value": "black"
    },
    {
      "name": "vermelho",
      "value": "red"
    },
    {
      "name": "rosa",
      "value": "pink"
    },
    {
      "name": "laranja",
      "value": "orange"
    },
    {
      "name": "amarelo",
      "value": "yellow"
    },
    {
      "name": "verde",
      "value": "green"
    },
    {
      "name": "magenta",
      "value": "magenta"
    },
    {
      "name": "ciano",
      "value": "cyan"
    },
    {
      "name": "azul",
      "value": "blue"
    },
    {
      "name": "azul_escuro",
      "value": "indigo"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_3d_de]=[
  [
    {
      "name": "rot_und_weiß",
      "value": "red_and_white"
    },
    {
      "name": "blau_und_weiß",
      "value": "blue_and_white"
    },
    {
      "name": "grün_und_weiß",
      "value": "green_and_white"
    },
    {
      "name": "grau_und_weiß",
      "value": "grey_and_white"
    },
    {
      "name": "orange_und_weiß",
      "value": "orange_and_white"
    },
    {
      "name": "lila_und_weiß",
      "value": "purple_and_white"
    },
    {
      "name": "gelb_und_weiß",
      "value": "yellow_and_white"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_3d_en]=[
  [
    {
      "name": "red_and_white",
      "value": "red_and_white"
    },
    {
      "name": "blue_and_white",
      "value": "blue_and_white"
    },
    {
      "name": "green_and_white",
      "value": "green_and_white"
    },
    {
      "name": "grey_and_white",
      "value": "grey_and_white"
    },
    {
      "name": "orange_and_white",
      "value": "orange_and_white"
    },
    {
      "name": "purple_and_white",
      "value": "purple_and_white"
    },
    {
      "name": "yellow_and_white",
      "value": "yellow_and_white"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_3d_fr]=[
  [
    {
      "name": "rouge_et_blanc",
      "value": "red_and_white"
    },
    {
      "name": "bleu_et_blanc",
      "value": "blue_and_white"
    },
    {
      "name": "vert_et_blanc",
      "value": "green_and_white"
    },
    {
      "name": "gris_et_blanc",
      "value": "grey_and_white"
    },
    {
      "name": "orange_et_blanc",
      "value": "orange_and_white"
    },
    {
      "name": "violet_et_blanc",
      "value": "purple_and_white"
    },
    {
      "name": "jaune_et_blanc",
      "value": "yellow_and_white"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_3d_pt_BR]=[
  [
    {
      "name": "vermelho_e_branco",
      "value": "red_and_white"
    },
    {
      "name": "azul_e_branco",
      "value": "blue_and_white"
    },
    {
      "name": "verde_e_branco",
      "value": "green_and_white"
    },
    {
      "name": "cinza_e_branco",
      "value": "grey_and_white"
    },
    {
      "name": "laranja_e_branco",
      "value": "orange_and_white"
    },
    {
      "name": "roxo_e_branco",
      "value": "purple_and_white"
    },
    {
      "name": "amarelo_e_branco",
      "value": "yellow_and_white"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_RdD_de]=[
  [
    {
      "name": "standard",
      "value": "default"
    },
    {
      "name": "spezial",
      "value": "special"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_RdD_en]=[
  [
    {
      "name": "default",
      "value": "default"
    },
    {
      "name": "special",
      "value": "special"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_RdD_fr]=[
  [
    {
      "name": "par_défaut",
      "value": "default"
    },
    {
      "name": "spécial",
      "value": "special"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_RdD_pt_BR]=[
  [
    {
      "name": "padrao",
      "value": "default"
    },
    {
      "name": "especial",
      "value": "special"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v1_de]=[
  [
    {
      "name": "schwarz_und_gold",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v1_en]=[
  [
    {
      "name": "black_and_gold",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v1_fr]=[
  [
    {
      "name": "noir_et_or",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v1_pt_BR]=[
  [
    {
      "name": "preto_e_dourado",
      "value": "black_and_gold"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v2_de]=[
  [
    {
      "name": "blau_und_silber",
      "value": "blue_and_silver"
    },
    {
      "name": "schwarz_und_gold",
      "value": "black_and_gold"
    },
    {
      "name": "blau_und_gold",
      "value": "blue_and_gold"
    },
    {
      "name": "grün_und_gold",
      "value": "green_and_gold"
    },
    {
      "name": "orange_und_silber",
      "value": "orange_and_silver"
    },
    {
      "name": "rot_und_gold",
      "value": "red_and_gold"
    },
    {
      "name": "lila_und_silber",
      "value": "purple_and_silver"
    },
    {
      "name": "schwarz_und_rot",
      "value": "black_and_red"
    },
    {
      "name": "regenbogen",
      "value": "rainbow"
    },
    {
      "name": "schwarz_und_silber",
      "value": "black_and_silver"
    },
    {
      "name": "pink_und_silber",
      "value": "pink_and_silver"
    },
    {
      "name": "gelb_und_braun",
      "value": "yellow_and_brown"
    },
    {
      "name": "lila_und_schwarz",
      "value": "purple_and_black"
    },
    {
      "name": "blau_und_schwarz",
      "value": "blue_and_black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v2_en]=[
  [
    {
      "name": "blue_and_silver",
      "value": "blue_and_silver"
    },
    {
      "name": "black_and_gold",
      "value": "black_and_gold"
    },
    {
      "name": "blue_and_gold",
      "value": "blue_and_gold"
    },
    {
      "name": "green_and_gold",
      "value": "green_and_gold"
    },
    {
      "name": "orange_and_silver",
      "value": "orange_and_silver"
    },
    {
      "name": "red_and_gold",
      "value": "red_and_gold"
    },
    {
      "name": "purple_and_silver",
      "value": "purple_and_silver"
    },
    {
      "name": "black_and_red",
      "value": "black_and_red"
    },
    {
      "name": "rainbow",
      "value": "rainbow"
    },
    {
      "name": "black_and_silver",
      "value": "black_and_silver"
    },
    {
      "name": "pink_and_silver",
      "value": "pink_and_silver"
    },
    {
      "name": "yellow_and_brown",
      "value": "yellow_and_brown"
    },
    {
      "name": "purple_and_black",
      "value": "purple_and_black"
    },
    {
      "name": "blue_and_black",
      "value": "blue_and_black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v2_fr]=[
  [
    {
      "name": "bleu_et_argent",
      "value": "blue_and_silver"
    },
    {
      "name": "noir_et_or",
      "value": "black_and_gold"
    },
    {
      "name": "bleu_et_or",
      "value": "blue_and_gold"
    },
    {
      "name": "vert_et_or",
      "value": "green_and_gold"
    },
    {
      "name": "orange_et_argent",
      "value": "orange_and_silver"
    },
    {
      "name": "rouge_et_or",
      "value": "red_and_gold"
    },
    {
      "name": "violet_et_argent",
      "value": "purple_and_silver"
    },
    {
      "name": "noir_et_rouge",
      "value": "black_and_red"
    },
    {
      "name": "arc_en_ciel",
      "value": "rainbow"
    },
    {
      "name": "noir_et_argent",
      "value": "black_and_silver"
    },
    {
      "name": "rose_et_argent",
      "value": "pink_and_silver"
    },
    {
      "name": "jaune_et_brun",
      "value": "yellow_and_brown"
    },
    {
      "name": "violet_et_noir",
      "value": "purple_and_black"
    },
    {
      "name": "bleu_et_noir",
      "value": "blue_and_black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_alies_v2_pt_BR]=[
  [
    {
      "name": "azul_e_prata",
      "value": "blue_and_silver"
    },
    {
      "name": "preto_e_dourado",
      "value": "black_and_gold"
    },
    {
      "name": "azul_e_dourado",
      "value": "blue_and_gold"
    },
    {
      "name": "verde_e_dourado",
      "value": "green_and_gold"
    },
    {
      "name": "laranja_e_prata",
      "value": "orange_and_silver"
    },
    {
      "name": "vermelho_e_dourado",
      "value": "red_and_gold"
    },
    {
      "name": "roxo_e_prata",
      "value": "purple_and_silver"
    },
    {
      "name": "preto_e_vermelho",
      "value": "black_and_red"
    },
    {
      "name": "arco_íris",
      "value": "rainbow"
    },
    {
      "name": "preto_e_prata",
      "value": "black_and_silver"
    },
    {
      "name": "rosa_e_prata",
      "value": "pink_and_silver"
    },
    {
      "name": "amarelo_e_marrom",
      "value": "yellow_and_brown"
    },
    {
      "name": "roxo_e_preto",
      "value": "purple_and_black"
    },
    {
      "name": "azul_e_preto",
      "value": "blue_and_black"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_knots_de]=[
  [
    {
      "name": "lila_dunkel",
      "value": "purple_dark"
    },
    {
      "name": "lila_weiß",
      "value": "purple_white"
    },
    {
      "name": "blau",
      "value": "blue"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_knots_en]=[
  [
    {
      "name": "purple_dark",
      "value": "purple_dark"
    },
    {
      "name": "purple_white",
      "value": "purple_white"
    },
    {
      "name": "blue",
      "value": "blue"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_knots_fr]=[
  [
    {
      "name": "violet_foncé",
      "value": "purple_dark"
    },
    {
      "name": "violet_blanc",
      "value": "purple_white"
    },
    {
      "name": "bleu",
      "value": "blue"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption[polyhedral_knots_pt_BR]=[
  [
    {
      "name": "roxo_escuro",
      "value": "purple_dark"
    },
    {
      "name": "roxo_claro",
      "value": "purple_white"
    },
    {
      "name": "azul",
      "value": "blue"
    },
    {
      "name": "none",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption_invalidStyle[de]=[
  [
    {
      "name": "Wähle zuerst einen Würfelbilderstil",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption_invalidStyle[en]=[
  [
    {
      "name": "Select the dice image style first",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption_invalidStyle[fr]=[
  [
    {
      "name": "Sélectionnez d'abord le style d'image des dés",
      "value": "none"
    }
  ]
]


de.janno.discord.bot.command.BaseCommandOptionsTest.autoCompleteColorOption_invalidStyle[pt_BR]=[
  [
    {
      "name": "Selecione o estilo da imagem do dado primeiro",
      "value": "none"
    }
  ]
]